homeassistant/components/zwave_js/__init__.py

```diff
@@ -20,40 +24,79 @@ async def async_setup(hass: HomeAssistant, config: dict):
     return True
 
 
+@callback
+def register_node_in_dev_reg(
+    entry: ConfigEntry,
+    dev_reg: device_registry.DeviceRegistry,
+    client: ZwaveClient,
+    node: ZwaveNode,
+) -> None:
+    """Register node in dev reg."""
+    dev_reg.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, client.driver.controller.home_id, node.node_id)},
+        sw_version=node.firmware_version,
+        name=node.name,
+        model=node.product_type,
+        manufacturer=str(node.manufacturer_id),
+    )
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Z-Wave JS from a config entry."""
-
-    # The ZwaveClient instance is stored in hass data for easy access from platforms.
     client = ZwaveClient(entry.data[CONF_URL], async_get_clientsession(hass))
-    hass.data[DOMAIN][entry.entry_id] = client
+    initialized = asyncio.Event()
 
     async def async_on_initialized():
         """Handle initial full state received."""
         # TODO: signal entities to update availability state
         LOGGER.info("Connection to Zwave JS Server initialized")
-        # register callbacks
-        client.driver.on("all nodes ready", async_on_ready)
+        initialized.set()
+
+        # This should be moved to the Node Ready command or something
+        # And then listen to new nodes too.
+        dev_reg = await device_registry.async_get_registry(hass)
+        for node in client.driver.controller.nodes.values():
+            register_node_in_dev_reg(entry, dev_reg, client, node)
```

Comment:
there should be a check here for the node state ?
and we indeed also need an event "node_ready"